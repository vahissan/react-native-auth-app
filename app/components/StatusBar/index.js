import React, { PureComponent } from "react";
import { StatusBar, Platform } from "react-native";
import Theme from "../../theme";

class StatusBarComponent extends PureComponent {
  render() {
    return Platform.select({
      ios: (
        <StatusBar
          translucent
          barStyle="dark-content"
        />
      ),
      android: (
        <StatusBar
          backgroundColor={Theme.colors.primaryColor}
          barStyle="light-content"
        />
      )
    });
  }
}

export default StatusBarComponent;
