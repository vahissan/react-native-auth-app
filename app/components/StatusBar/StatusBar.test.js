import React from 'react';
import { shallow } from 'enzyme';
import StatusBar from './index';

describe('StatusBar component', () => {
  jest.mock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'ios';
    return Platform;
  });
  
  it('should render propery', () => {
    const wrapper = shallow(<StatusBar />);
    expect(wrapper).toMatchSnapshot();
  });
});