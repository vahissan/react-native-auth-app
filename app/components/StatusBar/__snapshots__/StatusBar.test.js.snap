// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar component should render propery 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatusBarComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animated": false,
      "barStyle": "dark-content",
      "showHideTransition": "fade",
      "translucent": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": false,
        "barStyle": "dark-content",
        "showHideTransition": "fade",
        "translucent": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
